### PR TITLE
Fix e2e limits

### PR DESCRIPTION
**Pull request by ravisantoshgudimetla**
**What type of PR is this?**

 /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #
This should help in fixing #69989

**Special notes for your reviewer**:
Seems `clientset.CoreV1().Nodes().Patch(string(node.Name), types.StrategicMergePatchType, patchBytes)` doesn't update the node status which makes the all the nodes to have same allocatable values there by causing the test failure. Changed it to `updateStatus` which should help in resolving the bug.

/sig scheduling
/cc @bsalamat @Huang-Wei 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/scheduling/priorities.go [hunk 1]
```diff
@@ -457,20 +463,11 @@ func addRandomTaitToNode(cs clientset.Interface, nodeName string) *v1.Taint {
...
-	newData, err := json.Marshal(node)
-	if err != nil {
-		return err
-	}
-	patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldData, newData, v1.Node{})
-	if err != nil {
-		return err
-	}
-	_, err = c.CoreV1().Nodes().Patch(string(node.Name), types.StrategicMergePatchType, patchBytes)
+	_, err = c.CoreV1().Nodes().UpdateStatus(node)
```
Comment (Huang-Wei) [inline test/e2e/scheduling/priorities.go]: Is that to say, `Patch()` doesn't work for API fields in status?

File: test/e2e/scheduling/priorities.go [hunk 2]
```diff
@@ -281,14 +279,22 @@ var _ = SIGDescribe("SchedulerPriorities [Serial]", func() {
 		Expect(found).To(Equal(true))
 		nodeOriginalMemoryVal := nodeOriginalMemory.Value()
 		nodeOriginalCPUVal := nodeOriginalCPU.MilliValue()
-		err := updateNodeAllocatable(cs, nodeName, int64(10000), int64(12000))
+		err := updateNodeAllocatable(cs, nodeName, int64(10737418240), int64(12000))
```
Comment (Huang-Wei) [inline test/e2e/scheduling/priorities.go]: Any reason to set this particular value? Isn't `10000` (greater than incoming limit `5000m`) good enough?

File: test/e2e/scheduling/priorities.go (hunk 1 above)
Comment (ravisantoshgudimetla) [reply test/e2e/scheduling/priorities.go]: > Is that to say, Patch() doesn't work for API fields in status? 
Yeah, that's right. Patch wasn't working for updating cpu or memory. 

In order to test it, I added taints to node and applied patch, this worked while cpu and memory were not getting updated. So, I had to use UpdateStatus.

File: test/e2e/scheduling/priorities.go (hunk 2 above)
Comment (ravisantoshgudimetla) [reply test/e2e/scheduling/priorities.go]: This value is for memory not cpu

Comment (Huang-Wei) [reply test/e2e/scheduling/priorities.go]: sry, I misread... Let me rephrase: so the original mem val `10000` (greater than incoming limit `3000Mi`) should be put as a form `10737418240`, to represent 10Gi, right?

Comment (ravisantoshgudimetla) [reply test/e2e/scheduling/priorities.go]: Yes, that is right and that code path was not getting exercised as patch had not effect.

**Review by Huang-Wei — APPROVED** [review]
/lgtm

Thanks Ravi.
